src/keys.cc

```diff
@@ -49,7 +49,7 @@ Optional<Codepoint> Key::codepoint() const
         return '\n';
     if (*this == Key::Tab)
         return '\t';
-    if (*this == Key::Space)
+    if (*this == Key::Space or *this == shift(Key::Space))
```

Comment:
I guess with the same logic `<s-ret>` should also insert a newline in insert mode, as should `<s-tab>`.
Vim behaves that  way.

Reply:
But `<s-tab>` is backtab, and people expect it to unindent.

Reply:
> But `<s-tab>` is backtab, and people expect it to unindent.

Makes sense although it would be weird (for Kakoune) if <s-tab> in insert mode unindented.

---

Some other keys that don't work OOTB in terminals like foot are
<c-[> for Escape
<c-j> or <c-m> for Return
<c-i> for Tab

I'm not sure where to draw the line.

I guess <c-[> doubles as escape purely for historical reasons,
whereas <c-s> inserting a space makes sense even if it wasn't historical behavior.